### PR TITLE
Allow use of PV system data from PV Sites database

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

This pull request is intended to allow us to use PV data from the `pvsites` database. 

- When loading training data, the capacity of the PV system was previously taken to be the maximum observed output in the timeseries. When loading from the database, the capacity was set to the metadata value. The maximum observed and the metadata value can be quite different, and if we normalise the PV system data by them this will cause us trouble in production. 
    
  This pull request adds new variables to the PV system DataArray. Previously we only had `capacity_watt_power`. Now we have `observed_capacity_watt_power` and `metadata_capacity_watt_power` instead of this. This is more explicit and should help us avoid normalisation mistakes.

   - However, I'm wondering if it might be better to align the names more closely to the GSP capacity names - which are `nominal_capacity_mwp` and `effective_capacity_mwp`?

- Added `OpenPVFromPVSitesDBIterDataPipe` datapipe function (plus helper functions) to load from `pvsites` database.  The current `OpenPVFromPVDBIterDataPipe` datapipe function can only be used to load from the `pv` database. 
     
   I think we intend to move away from using the `pv` database, so I wasn't sure if we would want to replace `OpenPVFromPVDBIterDataPipe` to only load from `pvsites`. Currently, I have left the two functions alongside each other.

- Added new tests for `OpenPVFromPVSitesDBIterDataPipe`

- Added `ApplyPVDropoutIterDataPipe` to apply dropout specific for PV systems. Independently dropout each system and set a latency for each system.

- Added tests for `ApplyPVDropoutIterDataPipe`

- Add to PVNet datapipe to include PV system inputs

- Minor cleaning and refactoring

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ocf_datapipes/training/common.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def get_and_return_overlapping_time_periods_and_t0(used_datapipes: dict, key_for
     for i, key in enumerate(list(datapipes_to_return.keys())):
         datapipes_to_return[key + "_t0"] = t0_datapipes[i]
 
-    # Readd config for later
+    # Read config for later
```
Comment (jacobbieker) [inline ocf_datapipes/training/common.py:231]:
```suggestion
    # Re-add config for later
```

**Review by jacobbieker — APPROVED** [review]
LGTM! I would probably go with copying the GSP name conventions, just to keep it the same across those.
